objstore.go

```diff
@@ -62,7 +62,7 @@ type Bucket interface {
 
 	// Upload the contents of the reader as an object into the bucket.
 	// Upload should be idempotent.
-	Upload(ctx context.Context, name string, r io.Reader) error
+	Upload(ctx context.Context, name string, r io.Reader, opts ...ObjectUploadOption) error
```

Comment:
Why not simpler?

```suggestion
	Upload(ctx context.Context, name string, r io.Reader, opts ...UploadOption) error
```

Reply:
Looks like we already have this type: https://github.com/thanos-io/objstore/blob/3f8838fdd5a2c99bc3d771ef78816201e7d0b69e/objstore.go#L233.

It is used for the `UploadDir` function.